src/home_robot/home_robot/motion/stretch.py

```diff
@@ -487,16 +528,15 @@ def get_link_pose(self, link_name, q=None):
     def manip_fk(self, q: np.ndarray = None) -> Tuple[np.ndarray, np.ndarray]:
         """manipulator specific forward kinematics; uses separate URDF than the full-body fk() method"""
         assert q.shape == (self.dof,)
-        if self._ik_type == "pinocchio":
-            pin_pose = self._ros_pose_to_pinocchio(q)
-            ee_pos, ee_quat = self.manip_ik_solver.compute_fk(pin_pose)
-        elif self._ik_type == "pybullet":
-            ee_pos, ee_quat = self.fk(q)
+
+        if "pinocchio" in self._ik_type:
```

Comment:
IMO it would be preferable for the solver to do the conversion if necessary, but this change seemed like it might be larger than was suitable here.